aiida_quantumespresso/calculations/pw.py

```diff
@@ -134,6 +134,8 @@ def define(cls, spec):
         spec.exit_code(541, 'ERROR_SYMMETRY_NON_ORTHOGONAL_OPERATION',
             message='The variable cell optimization broke the symmetry of the k-points.')
 
+        spec.exit_code(551, 'ERROR_DEXX_IS_NEGATIVE', message='The code failed with negative dexx in the exchange calculation.')
```

Comment:
I was reserving the `5**` series for problems related to ionic calculations, such as `relax` and `md`. This is better suited as a 4xx. Since this is an error for hybrid functionals, let's reserve `46x` for those and make this one `461` (use `460` as the generic one in the future if needed).

Reply:
Looking at this section, I'm not sure if the comment ``# Significant errors but calculation can be used to restart`` applies to this error. Does that influence the choice of exit number, or is ``461`` good anyway?

Reply:
Yeah, the comment is a bit outdated and is not really what I intend anymore. Currently I maintain the following ranges, but that is still moving while we figure things out:
0 -  99: Reserved for `aiida-core` internal
100 - 199: Reserved for scheduler errors (will be implementing this soon, things like OOM, OOW and node failures
200 - 299: Extended input validation error (not sure if this is useful/necessary, but not everything is possible through spec validation, so some may have to be done in the process itself)
300 - 399: Critical errors where things really went tits up. Think any error that the program itself did not write. So typically output files will be missing, corrupt etc. for whatever reason.
400 - \inf: Any other error. Typically these will then be errors known and written by the program itself. Some of those can potentially be corrected by the calling work chain and some not. I am now using sub ranges for different types, e.g. 400 - 499 for electronic type calculations 'scf/nscf/bands' and 500 - 599 for ionic relaxation 'relax/vc-relax'. Mayb at some point 600 - 699 for 'md/vc-md'.

Reply:
Makes sense. Should I remove the comment?

Reply:
Let's leave it for now. Pretty soon I will be doing a big update of all calculations to reflect a consistent use of exit codes